test/unit/org/apache/cassandra/cql3/validation/operations/SelectTest.java

```diff
@@ -3191,4 +3191,20 @@ public void testCreatingUDFWithSameNameAsBuiltin_FullyQualifiedFunctionNameWorks
         execute("INSERT INTO %s (k1, k2) VALUES (uuid(), 'k2')");
         assertRowCount(execute("SELECT system.token(k1, k2) FROM %s"), 1);
     }
+
+    @Test
+    public void testLikeQueryWithoutFiltering() throws Throwable
+    {
+        createTable("CREATE TABLE %s (k1 uuid, k2 text, PRIMARY KEY (k1, k2))");
+        execute("INSERT INTO %s (k1, k2) VALUES(uuid(), 'John Doe')");
+        assertRows(execute("SELECT k2 from %s where k2 like '%%Doe'"), row("John Doe"));
+    }
```

Comment:
That test should not pass. For filtering on partition columns the `needFiltering` methods in `PartitionKeySingleRestrictionSet` need to be modified to return `true` when LIKE predicate are used.

Reply:
Thanks, this test is actually passing now, let me also check the `PartitionKeySingleRestrictionSet `